### PR TITLE
[Keyboard] Add keyboard Reviung41

### PR DESCRIPTION
**Pull request by gtips**
Add keyboard Reviung41.

## Description
The REVIUNG41 is 41-key column staggered keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/reviung41/config.h [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
+#define DEVICE_VER 0x0001
+#define MANUFACTURER % YOUR_NAME %
```
Comment (noroadsleft) [inline keyboards/reviung41/config.h]:
```suggestion
#define MANUFACTURER gtips
```

File: keyboards/reviung41/config.h [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+#define PRODUCT % KEYBOARD %
```
Comment (noroadsleft) [inline keyboards/reviung41/config.h]:
```suggestion
#define PRODUCT reviung41
```

File: keyboards/reviung41/config.h [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
```
Comment (noroadsleft) [inline keyboards/reviung41/config.h]: We're trying to get all the USB Device descriptors unique within QMK.

```suggestion
#define PRODUCT_ID 0xDCCB
```

File: keyboards/reviung41/keymaps/default/config.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+//   #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//   #define RGBLIGHT_EFFECT_RGB_TEST
+//   #define RGBLIGHT_EFFECT_ALTERNATING
+// /*== customize breathing effect ==*/
+//   /*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+//   #define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+//   /*==== use exp() and sin() ====*/
+//   #define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+//   #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+#endif
```
Comment (noroadsleft) [inline keyboards/reviung41/keymaps/default/config.h]: Lines 21-47 should all be in `keyboards/reviung41/config.h`.

File: keyboards/reviung41/keymaps/default/rules.mk [hunk 5]
```diff
@@ -0,0 +1 @@
+RGBLIGHT_ENABLE = yes
```
Comment (noroadsleft) [inline keyboards/reviung41/keymaps/default/rules.mk]: This should be configured at the keyboard level in `keyboards/reviung41/rules.mk`. Please delete this file:

```
git rm -- keyboards/reviung41/keymaps/default/rules.mk
```

File: keyboards/reviung41/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# reviung41
+
+![reviung41](https://github.com/gtips/reviung/blob/master/reviung41/image/REVIUNG41-prototype.jpg)  
+
+The REVIUNG41 is 41-key column staggered keyboard.  
+
+Keyboard Maintainer: [gtips](https://github.com/gtips)  
+Hardware Supported: The REVIUNG41 PCBs, Pro Micro supported  
+Hardware Availability: [PCB & Case Data](https://github.com/gtips/reviung/tree/master/reviung41)
```
Comment (noroadsleft) [inline keyboards/reviung41/readme.md]: Please change Lines 7-9 to:

```Markdown
* Keyboard Maintainer: [gtips](https://github.com/gtips)
* Hardware Supported: REVIUNG41 PCBs, Pro Micro supported
* Hardware Availability: [PCB & Case Data](https://github.com/gtips/reviung/tree/master/reviung41)
```

File: keyboards/reviung41/reviung41.c [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  return process_record_user(keycode, record);
+}
+
+void led_set_kb(uint8_t usb_led) {
+  // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+  led_set_user(usb_led);
+}
+
+*/
```
Comment (noroadsleft) [inline keyboards/reviung41/reviung41.c]: Lines 18-51 can all be deleted.

File: keyboards/reviung41/rules.mk [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (noroadsleft) [inline keyboards/reviung41/rules.mk]: Please delete Lines 14-22.

File: keyboards/reviung41/rules.mk [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (noroadsleft) [inline keyboards/reviung41/rules.mk]: Please change Lines 26-42 to:

```Makefile
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

This removes the firmware size impact notes, aligns the comments, and enables RGB Underglow at the keyboard level.

Comment (gtips) [reply keyboards/reviung41/rules.mk]: Thanks

File: keyboards/reviung41/keymaps/default/config.h [hunk 10]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 gtips
```
Comment (fauxpark) [inline keyboards/reviung41/keymaps/default/config.h]: This file can be deleted.

File: keyboards/reviung41/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  
+  [_ADJUST] = LAYOUT_reviung41(
+    RGB_VAI,   RGB_SAI, RGB_HUI,  RGB_MOD,  XXXXXXX,   RGB_TOG,            XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+    RGB_VAD,   RGB_SAD, RGB_HUD,  RGB_RMOD, XXXXXXX,   XXXXXXX,            XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+    XXXXXXX,   XXXXXXX, XXXXXXX,  XXXXXXX,  XXXXXXX,   XXXXXXX,            RESET,    XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+                                            _______,   _______,  XXXXXXX,  _______,  _______
+  ),
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (fauxpark) [inline keyboards/reviung41/keymaps/default/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: keyboards/reviung41/readme.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+# reviung41
+
+![reviung41](https://github.com/gtips/reviung/blob/master/reviung41/image/REVIUNG41-prototype.jpg)  
+
+The REVIUNG41 is 41-key column staggered keyboard.
```
Comment (fauxpark) [inline keyboards/reviung41/readme.md]:
```suggestion
The REVIUNG41 is 41-key column staggered keyboard.
```

File: keyboards/reviung41/readme.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
+# reviung41
+
+![reviung41](https://github.com/gtips/reviung/blob/master/reviung41/image/REVIUNG41-prototype.jpg)  
+
+The REVIUNG41 is 41-key column staggered keyboard.  
+
+* Keyboard Maintainer: [gtips](https://github.com/gtips)
+* Hardware Supported: REVIUNG41 PCBs, Pro Micro supported
+* Hardware Availability: [PCB & Case Data](https://github.com/gtips/reviung/tree/master/reviung41)
```
Comment (fauxpark) [inline keyboards/reviung41/readme.md]:
```suggestion
* Hardware Availability: [PCB & Case Data](https://github.com/gtips/reviung/tree/master/reviung41)
```

File: keyboards/reviung41/readme.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
+# reviung41
+
+![reviung41](https://github.com/gtips/reviung/blob/master/reviung41/image/REVIUNG41-prototype.jpg)
```
Comment (fauxpark) [inline keyboards/reviung41/readme.md]:
```suggestion
![reviung41](https://github.com/gtips/reviung/blob/master/reviung41/image/REVIUNG41-prototype.jpg)
```

Comment (gtips) [reply keyboards/reviung41/readme.md]: Thanks.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**